WordPress/Classes/ViewRelated/NUX/SignupEpilogueTableViewController.swift

```diff
@@ -0,0 +1,241 @@
+        return Constants.headerFooterHeight
+    }
+
+    override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableViewAutomaticDimension
+    }
+
+    override func tableView(_ tableView: UITableView, estimatedHeightForFooterInSection section: Int) -> CGFloat {
+        return Constants.headerFooterHeight
+    }
+
+    override func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        if section == TableSections.password {
+            return UITableViewAutomaticDimension
+        }
+        return 0
+    }
+
+}
+
+// MARK: - Private Extension
+
+private extension SignupEpilogueTableViewController {
+
+    func configureTable() {
+        let headerFooterNib = UINib(nibName: CellNibNames.sectionHeaderFooter, bundle: nil)
+        tableView.register(headerFooterNib, forHeaderFooterViewReuseIdentifier: CellIdentifiers.sectionHeaderFooter)
+
+        let cellNib = UINib(nibName: CellNibNames.signupEpilogueCell, bundle: nil)
+        tableView.register(cellNib, forCellReuseIdentifier: CellIdentifiers.signupEpilogueCell)
+
+        let userInfoNib = UINib(nibName: CellNibNames.epilogueUserInfoCell, bundle: nil)
+        tableView.register(userInfoNib, forCellReuseIdentifier: CellIdentifiers.epilogueUserInfoCell)
+
+        WPStyleGuide.configureColors(for: view, andTableView: tableView)
+
+        // remove empty cells
+        tableView.tableFooterView = UIView()
+    }
+
+    func getUserInfo() {
+        if loginFields.meta.socialService == .google {
+            showPassword = false
+            var userInfo = LoginEpilogueUserInfo()
+            userInfo.email = loginFields.emailAddress
+            userInfo.username = loginFields.username
+
+            if let fullName = googleUser?.profile.name {
+                userInfo.fullName = fullName
+            }
+
+            epilogueUserInfo = userInfo
+        } else {
+            let service = AccountService(managedObjectContext: ContextManager.sharedInstance().mainContext)
+            if let account = service.defaultWordPressComAccount() {
+                epilogueUserInfo = LoginEpilogueUserInfo(account: account)
+            }
+        }
+    }
+
+    func getEpilogueCellFor(cellType: EpilogueCellType) -> SignupEpilogueCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: CellIdentifiers.signupEpilogueCell) as? SignupEpilogueCell else {
+            fatalError("Failed to get epilogue cell")
+        }
+
+        switch cellType {
+        case .displayName:
+            cell.configureCell(labelText: NSLocalizedString("Display Name", comment: "Display Name label text."), fieldValue: epilogueUserInfo?.fullName)
+            cell.cellField.addTarget(self, action: #selector(displayNameDidChange(_:)), for: .editingChanged)
+            cell.cellField.addTarget(self, action: #selector(displayNameChanged(_:)), for: .editingDidEnd)
```

Comment:
I'd prefer not to use `addTarget` on the text field, and instead use `UITextFieldDelegate` in the cell itself. The main reason is so that we can avoid needing `@objc` on these methods.

You can see an example of what I mean in `SiteCreationDomainSearchTableViewCell`